src/omnisharp/experimentalOmnisharp.PackageCreator.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Package } from "../packages";
+
+export function GetPackagesFromVersion(version: string, runTimeDependencies: Package[], serverUrl: string, installPath: string): Package[] {
+    if (!version) {
+        throw new Error('Invalid version');
+    }
+
+    let versionPackages = new Array<Package>();
+    for (let inputPackage of runTimeDependencies) {
+        if (inputPackage.experimentalPackageId) {
+            versionPackages.push(GetExperimentPackage(inputPackage, serverUrl, version, installPath));
+        }
+    }
+
+    return versionPackages;
+}
+
+export function GetExperimentPackage(inputPackage: Package, serverUrl: string, version: string, installPath: string): Package {
+    let installBinary: string;
+    if (inputPackage.experimentalPackageId == "win-x86" || inputPackage.experimentalPackageId == "win-x64") {
```

Comment:
You're saying we should update the package.json schema for each package to list its architecture(?) as `experimentalPackageId`?

Reply:
Yes that way we dont have to parse the url or any other parameter to get the os and architecture to form the new download urls